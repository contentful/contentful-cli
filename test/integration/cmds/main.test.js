const nixt = require('nixt')
const { join } = require('path')

const packageVersion = require('../../../package.json').version
const bin = join(__dirname, './../../../', 'bin')

const app = () => {
  return nixt({ newlines: true })
    .cwd(bin)
    .base('./contentful.js ')
    .clone()
}

test('should return code 1 when errors exist no args', done => {
  app()
    .run('')
    .code(1)
    .expect(result => {
      const resultText = result.stderr.trim()
      expect(resultText).toMatchSnapshot('help data is incorrect')
    })
    .end(done)
})

test('should print help message', done => {
  app()
    .run('--help')
    .code(0)
    .expect(result => {
      const resultText = result.stdout.trim()
      expect(resultText).toMatchSnapshot('help data is incorrect')
    })
    .end(done)
})

test('should print help message on shortcut', done => {
  app()
    .run('-h')
    .code(0)
    .stdout(/Usage: contentful <cmd> \[args\]/)
    .expect(result => {
      const resultText = result.stdout.trim()
      expect(resultText).toMatchSnapshot('help data is incorrect')
    })
    .end(done)
})

test('should print help message on wrong subcommand', done => {
  app()
    .run('lolbar')
    .code(1)
    .expect(result => {
      const resultText = result.stderr.trim()
      expect(resultText).toMatchSnapshot('help data is incorrect')
    })
    .end(done)
})

test('should print version number', done => {
  app()
    .run('--version')
    .code(0)
    .stdout(packageVersion)
    .end(err => {
      expect(err).toBeFalsy()
      done()
    })
})
