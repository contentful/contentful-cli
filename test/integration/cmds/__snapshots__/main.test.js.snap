// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should print help message on shortcut: help data is incorrect 1`] = `
"Usage: contentful <cmd> [args]

Commands:
  boilerplate   Download a boilerplate
  config        Manage and list your configs
  content-type  Manage and list your space content types           [aliases: ct]
  extension     Manage and list your extensions
  guide         A guide introducing basic concepts of working with Contentful
  login         Login to Contentful
  logout        Logout from Contentful
  organization  Manage and list your organizations
  space         Manage and list your spaces

Options:
  -h, --help     Show help                                             [boolean]
  -v, --version  Show current version                                  [boolean]

Copyright 2019 Contentful"
`;

exports[`should print help message on wrong subcommand: help data is incorrect 1`] = `
"Usage: contentful <cmd> [args]

Commands:
  boilerplate   Download a boilerplate
  config        Manage and list your configs
  content-type  Manage and list your space content types           [aliases: ct]
  extension     Manage and list your extensions
  guide         A guide introducing basic concepts of working with Contentful
  login         Login to Contentful
  logout        Logout from Contentful
  organization  Manage and list your organizations
  space         Manage and list your spaces

Options:
  -h, --help     Show help                                             [boolean]
  -v, --version  Show current version                                  [boolean]

Copyright 2019 Contentful
Please specify a command."
`;

exports[`should print help message: help data is incorrect 1`] = `
"Usage: contentful <cmd> [args]

Commands:
  boilerplate   Download a boilerplate
  config        Manage and list your configs
  content-type  Manage and list your space content types           [aliases: ct]
  extension     Manage and list your extensions
  guide         A guide introducing basic concepts of working with Contentful
  login         Login to Contentful
  logout        Logout from Contentful
  organization  Manage and list your organizations
  space         Manage and list your spaces

Options:
  -h, --help     Show help                                             [boolean]
  -v, --version  Show current version                                  [boolean]

Copyright 2019 Contentful"
`;

exports[`should return code 1 when errors exist no args: help data is incorrect 1`] = `
"Usage: contentful <cmd> [args]

Commands:
  boilerplate   Download a boilerplate
  config        Manage and list your configs
  content-type  Manage and list your space content types           [aliases: ct]
  extension     Manage and list your extensions
  guide         A guide introducing basic concepts of working with Contentful
  login         Login to Contentful
  logout        Logout from Contentful
  organization  Manage and list your organizations
  space         Manage and list your spaces

Options:
  -h, --help     Show help                                             [boolean]
  -v, --version  Show current version                                  [boolean]

Copyright 2019 Contentful
Please specify a command."
`;
